### PR TITLE
[curl] Mitigate issue #14834: port broken on Linux

### PR DESCRIPTION
**Pull request by klalumiere**
**Describe the pull request**

- What does your PR fix?

Mitigate issue #14834: port broken on Linux
    
The more complete fix will come from the pull request #14807. However, since curl on Linux is broken right now, I think it might be worth it to merge this branch quickly so that curl will work again on Linux.

- Which triplets are supported/not supported? Have you updated the CI baseline? Nothing changed here

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? Yes

### CODE REVIEW

File: ports/curl/portfile.cmake [hunk 1]
```diff
@@ -182,7 +182,7 @@ endif()
 if(VCPKG_TARGET_IS_WINDOWS)	
     vcpkg_fixup_pkgconfig()
 elseif(VCPKG_TARGET_IS_LINUX)
-    vcpkg_fixup_pkgconfig(SYSTEM_LIBRARIES pthread dl c)
+    # vcpkg_fixup_pkgconfig(SYSTEM_LIBRARIES pthread dl c)
```
Comment (klalumiere) [inline ports/curl/portfile.cmake:185]: This was introduced ~10 days ago in #14350. The fix to make this work is developed in pull request #14807.

**Comment by JackBoosY** [comment]
No, this will break the port using curl pkgconig.

**Comment by klalumiere** [comment]
> No, this will break the port using curl pkgconig.

Well, yes and no 🙂 . As explained above in my comment, this line was added ~10 days ago in #14350. It only affect the behaviour on Linux, and right now, because of this line, the port doesn't work at all on Linux 
(neither does anything that depends, directly or transitively, on curl). Hence, I'd argue that this is a good first step , and then your pull request #14807 will provide a more complete fix that allows to use pkgconfig on Linux. 🙂

**Comment by klalumiere** [comment]
> No, this will break the port using curl pkgconig.

Well, yes and no 🙂 . As explained above in my comment, this line was added ~10 days ago in #14350. It only affect the behaviour on Linux, and right now, because of this line, the port doesn't work at all on Linux 
(neither does anything that depends, directly or transitively, on curl). Hence, I'd argue that this is a good first step , and then your pull request #14807 will provide a more complete fix that allows to use pkgconfig on Linux. 🙂

**Comment by klalumiere** [comment]
I was doing some tests for pull request #14807, and in the midst of these tests, I've realized that for I reason that I don't understand, I'm not able to reproduce the problem anymore. I guess that's a good news, although I'm curious and I'd like to understand what happen :laughing: . I'm closing and I'm going to leave a comment on issue #14834 to see if the issue is solved for other people to. I'm going to continue to follow #14807 (I'll add a comment there in ~1h if everything goes well).

**Comment by klalumiere** [comment]
I was doing some tests for pull request #14807, and in the midst of these tests, I've realized that for I reason that I don't understand, I'm not able to reproduce the problem anymore. I guess that's a good news, although I'm curious and I'd like to understand what happen :laughing: . I'm closing and I'm going to leave a comment on issue #14834 to see if the issue is solved for other people to. I'm going to continue to follow #14807 (I'll add a comment there in ~1h if everything goes well).

**Pull request closed without merging**